trustfall_core/src/schema/adapter/mod.rs

```diff
@@ -73,6 +73,61 @@ impl<'a> SchemaAdapter<'a> {
     pub fn schema_text() -> &'static str {
         include_str!("./schema.graphql")
     }
+
+    pub fn vertex_type_iter(
+        &self,
+        candidate_value_for_name: Option<CandidateValue<&FieldValue>>,
+    ) -> VertexIterator<'a, SchemaVertex<'a>> {
+        let root_query_type = self.schema.query_type_name();
+
+        if let Some(CandidateValue::Single(FieldValue::String(name_wanted))) =
+            candidate_value_for_name
+        {
+            let name_wanted = name_wanted.as_ref();
+            if let Some(exact_wanted) = self
+                .schema
+                .vertex_types
+                .get(name_wanted)
+                .filter(move |v| v.name.node != root_query_type)
+            {
+                Box::new(std::iter::once(SchemaVertex::VertexType(VertexType::new(exact_wanted))))
+            } else {
+                Box::new(std::iter::empty())
+            }
+        } else if let Some(CandidateValue::Multiple(possibilities)) = candidate_value_for_name {
+            let possibilities_as_owned_strings = possibilities
+                .iter()
+                .map(|el| el.as_str().expect("for name possibility to be a string").to_string())
```

Comment:
Oh with the new `FieldValue` representation, there's a better option here: we could add an `as_arc_str() -> Option<Arc<str>>` method on `FieldValue` and save on the deep-clone of the string and probably _also_ on the vector as well.

Wanna open a PR for that, or should I?

Reply:
Merged that PR: #414

Reply:
Sounds good